images/ami_kms_key.tf

```diff
@@ -0,0 +1,81 @@
+# ------------------------------------------------------------------------------
+# Create the KMS key for encrypting AMIs in the Images account
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "ami_kms_doc" {
+  statement {
+    sid = "Enable IAM User Permissions"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.images.account_id}:root"]
+    }
+
+    actions = [
+      "kms:*",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid = "Allow access for Key Administrators"
+
+    principals {
+      type = "AWS"
+      # This role needs to be created before the key is provisioned,
+      # so we can't use aws_iam_role.administerkmskeys_role.arn here.
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.images.account_id}:role/${var.administerkmskeys_role_name}"]
+    }
+
+    actions = [
+      "kms:CancelKeyDeletion",
+      "kms:Create*",
+      "kms:Delete*",
+      "kms:Describe*",
+      "kms:Disable*",
+      "kms:Enable*",
+      "kms:Get*",
+      "kms:List*",
+      "kms:Put*",
+      "kms:Revoke*",
+      "kms:ScheduleKeyDeletion",
+      "kms:TagResource",
+      "kms:UntagResource",
+      "kms:Update*",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid = "Allow use of the key"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["${aws_iam_role.ec2amicreate_role.arn}"]
+    }
+
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+      "kms:Encrypt",
+      "kms:GenerateDataKey*",
+      "kms:ReEncrypt*",
+    ]
+
+    resources = ["*"]
```

Comment:
It might be possible to limit this to only this particular KMS key. I'm unsure about this.

Reply:
I'm not sure how we would do that since we don't have the key ARN until the key is created, and we cannot create the key without this key policy.  Maybe we could do this in 2 steps:
1. Create the key with a `["*"]` key policy
1. Update the key policy to include the key's ARN

Reply:
I was hoping we could do something clever with the alias, but it looks like aliases cannot be used within policies :(